File: pset6/caesar.py
from sys import argv
from cs50 import get_string


if len(argv) != 2:
    exit("Usage: python caesar.py k")

if int(argv[1]) < 0:
    exit("Usage: python caesar.py k")

key = int(argv[1])
plaintext = get_string("plaintext: ")
ciphertext = ""

for s in plaintext:
    if s.isalpha():
        changing_latter = chr((ord(s.lower()) - 97 + key)%26 + 97)
        if s.isupper():
            ciphertext += changing_latter.upper()
        else:
            ciphertext += changing_latter
    else:
        ciphertext += s
print (f"ciphertext:", ciphertext)

File: pset2/ceasar/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <stdlib.h>

int main(int argc, string argv[])
{
    if (argc == 2)
    {
        string skey = argv[1];
        int only_int = 1;
        for (int i = 0; i < strlen(skey); i++)
        {
            if ( isdigit(skey[i]) == 0)
            {
                only_int = 0;
            }
        }
        if (only_int == 1)
        {
            int ikey = atoi(skey);
            string text = get_string("plaintext: ");
            printf("ciphertext:");
            for (int i = 0; i < strlen(text); i++)
            {
                if (isalpha(text[i]) > 0)
                {
                    int lower_latter = 1;
                    if (islower(text[i]) == 0)
                    {
                        char changing_latter = (tolower(text[i]) - 97 + ikey)%26 + 97;
                        printf("%c", toupper(changing_latter));
                    }
                    else
                    {
                        char changing_latter = (text[i] - 97 + ikey)%26 + 97;
                        printf("%c", changing_latter);

                    }
                }
                else
                {
                    printf("%c", text[i]);
                }
            }
            printf("\n");
        }
    }
    else
    {
        printf("Usage: ./caesar key\n");
        return 1;
    }
}